packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -552,9 +552,10 @@ export class Stack extends Construct implements ITaggable {
     const s3Filename = Fn.select(1, Fn.split(cxapi.ASSET_PREFIX_SEPARATOR, encodedKey));
     const objectKey = `${s3Prefix}${s3Filename}`;
 
-    const s3Url = `https://s3.${this.region}.${this.urlSuffix}/${bucketName}/${objectKey}`;
+    const httpUrl = `https://s3.${this.region}.${this.urlSuffix}/${bucketName}/${objectKey}`;
+    const s3ObjectUrl = `s3://${bucketName}/${objectKey}`;
```

Comment:
Can we add a unit test for this? I see there is an integration test in the `s3-assets` module that indirectly checks this, but i'd prefer a more explicit test in the `core` module as well.

Reply:
Sure!

I'm not quite sure how to set up the expectation though. The returned assetLocation contains tokens:
```ts
const assetLocation = stack.addFileAsset(...);
...
test.deepEqual(assetLocation, {
  bucketName: '${Token[TOKEN.79]}',
  objectKey: '${Token[TOKEN.82]}${Token[TOKEN.84]}',
  httpUrl: 'https://s3.${Token[AWS::Region.4]}.${Token[AWS::URLSuffix.1]}/${Token[TOKEN.79]}/${Token[TOKEN.82]}${Token[TOKEN.84]}',
  s3ObjectUrl: 's3://${Token[TOKEN.79]}/${Token[TOKEN.82]}${Token[TOKEN.84]}',
  s3Url: 'https://s3.${Token[AWS::Region.4]}.${Token[AWS::URLSuffix.1]}/${Token[TOKEN.79]}/${Token[TOKEN.82]}${Token[TOKEN.84]}',
});
```
Is there a better way to test this?

Reply:
@iliapolo Any feedback on the above? Does a test like this make sense? Is there a better way to test this?